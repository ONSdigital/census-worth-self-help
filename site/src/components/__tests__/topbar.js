import React from "react"
import renderer from "react-test-renderer"

import TopBar from "../topbar"
import { render, fireEvent } from "react-testing-library"

import { navigate } from "@reach/router"

jest.mock('@reach/router', () => ({
  navigate: jest.fn(),
}))

describe("TopBar", () => {

  let updateFunction = jest.fn()

  const searchObject = {
    updateFunction: updateFunction,
    query: "searchString"
  }

  it("renders correctly", () => {
    const tree = renderer
      .create(<TopBar/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly with searchbox", () => {
    const tree = renderer
      .create(<TopBar searchObject={searchObject} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("has working search box", () => {
    const { getByTestId, getAllByTestId, queryByTestId } = render(<TopBar searchObject={searchObject} />);
  	
    const searchBox = getByTestId('search-box')

  	expect(searchBox.value).toEqual("searchString")

  	fireEvent.change(searchBox, { target: { value: 'TEST VALUE' } });

  	expect(updateFunction).toHaveBeenCalled();
  })

  it("menu links return to original page", () => {

    // create a fake window object with the pathname parameter and the redirect parameter.
    global.window = Object.create(window);
    Object.defineProperty(window, 'location', {
      value: {
        pathname: "testtest",
        search: "?redirect=" + encodeURIComponent("/testtest/")
      }
    });

    // if you are on page testtest and click menu:
    let { getByText } = render(<TopBar />);
    
    const menuButton = getByText('Menu')
    fireEvent.click(menuButton);

    // you get sent to the menu page with redirect link in the url
    expect(navigate).toHaveBeenCalledTimes(1)
    expect(navigate).toHaveBeenCalledWith('/menu/?redirect=testtest')

    // if you have the redirect link in the url and click close
    getByText = render(<TopBar backButton={true} />).getByText;
    
    const closeButton = getByText('Close')
    fireEvent.click(closeButton);

    // you return to the original page
    expect(navigate).toHaveBeenCalledTimes(2)
    expect(navigate).toHaveBeenCalledWith('/testtest/')
  })
})