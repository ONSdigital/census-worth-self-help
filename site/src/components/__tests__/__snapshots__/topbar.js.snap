// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar renders correctly 1`] = `
<header
  css={
    Object {
      "map": undefined,
      "name": "181ra4m",
      "next": undefined,
      "styles": "
        width: 100vw;
        
    background: linear-gradient(to left, rgb(0,61,89), #33637a);
  ;
      ",
    }
  }
>
  <div
    css={
      Object {
        "map": undefined,
        "name": "1thgah9",
        "next": undefined,
        "styles": "
            display: flex;
          ",
      }
    }
  >
    <div
      className="Header-Title-Style"
      css={
        Object {
          "map": undefined,
          "name": "17ih1bc",
          "next": undefined,
          "styles": "
              
    display: flex;
    justify-content: center;
    flex-direction: column;
  
              
    padding: 15px;
  
              flex-grow: 1;
            ",
        }
      }
      onClick={[Function]}
    >
      Self Help Facility
    </div>
    <div
      className="Header-button-Style"
      css={
        Object {
          "map": undefined,
          "name": "1sm5jo",
          "next": undefined,
          "styles": "
        
    display: flex;
    justify-content: center;
    flex-direction: column;
  
        padding: 0px 10px;
        background-color: rgb(255, 255, 255, 0.2);
        border-radius: 4px;
        margin: 10px 10px 10px 0px;
        min-width: 20px;
        object-fit: contain;
        text-align: center;
        height: 40px;
      ",
        }
      }
      data-testid="topbarlink-button"
      onClick={[Function]}
    >
      <div
        css={
          Object {
            "map": undefined,
            "name": "1dbjmka",
            "next": undefined,
            "styles": "
                  font-size: 19px;
                ",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-search fa-w-16 "
          data-icon="search"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
    </div>
    <div
      className="Header-button-Style"
      css={
        Object {
          "map": undefined,
          "name": "1sm5jo",
          "next": undefined,
          "styles": "
        
    display: flex;
    justify-content: center;
    flex-direction: column;
  
        padding: 0px 10px;
        background-color: rgb(255, 255, 255, 0.2);
        border-radius: 4px;
        margin: 10px 10px 10px 0px;
        min-width: 20px;
        object-fit: contain;
        text-align: center;
        height: 40px;
      ",
        }
      }
      data-testid="topbarlink-button"
      onClick={[Function]}
    >
      Menu
    </div>
  </div>
</header>
`;

exports[`TopBar renders correctly with searchbox 1`] = `
<header
  css={
    Object {
      "map": undefined,
      "name": "181ra4m",
      "next": undefined,
      "styles": "
        width: 100vw;
        
    background: linear-gradient(to left, rgb(0,61,89), #33637a);
  ;
      ",
    }
  }
>
  <div
    css={
      Object {
        "map": undefined,
        "name": "1thgah9",
        "next": undefined,
        "styles": "
            display: flex;
          ",
      }
    }
  >
    <div
      css={
        Object {
          "map": undefined,
          "name": "1uk2jmg",
          "next": undefined,
          "styles": "
              padding: 10px 15px 14px 15px;
              flex-grow: 1;
              display: flex;
            ",
        }
      }
    >
      <div
        css={
          Object {
            "map": undefined,
            "name": "10cznun",
            "next": undefined,
            "styles": "
                
    display: flex;
    justify-content: center;
    flex-direction: column;
  
                position: relative;
                left: 5px;
              ",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-search fa-w-16 "
          css={
            Object {
              "map": undefined,
              "name": "5duw79",
              "next": undefined,
              "styles": "
                  position: absolute;
                  color: #6e6e6e;
                ",
            }
          }
          data-icon="search"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        style={
          Object {
            "border": 0,
            "clip": "rect(0 0 0 0)",
            "height": "1px",
            "margin": "-1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "absolute",
            "width": "1px",
          }
        }
      >
        <label
          htmlFor="search-box"
        >
          Breadcrumb Select Box
        </label>
      </div>
      <input
        autoFocus={true}
        className="Notification-heading-Style"
        css={
          Object {
            "map": undefined,
            "name": "1f3qd7n",
            "next": undefined,
            "styles": "
                border: 0;
                border-radius: 16px;
                background-color: rgb(250, 250, 250);
                flex-grow: 1;
                padding-left: 25px;
              ",
          }
        }
        data-testid="search-box"
        id="search-box"
        maxLength="60"
        onChange={[MockFunction]}
        type="text"
        value="searchString"
      />
    </div>
    <div
      className="Header-button-Style"
      css={
        Object {
          "map": undefined,
          "name": "1sm5jo",
          "next": undefined,
          "styles": "
        
    display: flex;
    justify-content: center;
    flex-direction: column;
  
        padding: 0px 10px;
        background-color: rgb(255, 255, 255, 0.2);
        border-radius: 4px;
        margin: 10px 10px 10px 0px;
        min-width: 20px;
        object-fit: contain;
        text-align: center;
        height: 40px;
      ",
        }
      }
      data-testid="topbarlink-button"
      onClick={[Function]}
    >
      Menu
    </div>
  </div>
</header>
`;
